plugin/type_uuid/item_uuidfunc.cc

```diff
@@ -44,3 +45,36 @@ bool Item_func_uuid::val_native(THD *, Native *to)
   my_uuid((uchar*)to->ptr());
   return 0;
 }
+
+const Type_handler *Item_func_uuid_v4::type_handler() const
+{
+  return Type_handler_uuid_new::singleton();
+}
+
+String *Item_func_uuid_v4::val_str(String *str)
+{
+  DBUG_ASSERT(fixed());
+  str->alloc(uuid_len()+1);
+  str->length(uuid_len());
+  str->set_charset(collation.collation);
+
+  uchar buf[MY_UUID_SIZE];
+  if (!my_uuid_v4(buf)) {
+    my_error(ER_INTERNAL_ERROR, MYF(0),
+    "Failed to generate a random value for UUIDv4");
+  }
```

Comment:
Have you tested to see what it takes to get this? I suspect once a certain amount of system entropy has been consumed but I haven't looked at the openssl implementation.

Reply:
The docs says:

> If the entropy source fails or is not available, the CSPRNG will enter an error state and refuse to generate random bytes

I generated a big amount of UUIDv4 but I could not get to the point of making it fail on my machine.
I hard-coded it to always fail and checked that the behavior is as expected in case of failure.
If you have any idea on how to test this let me know.

Reply:
That's good enough. Just making sure a big UUIDv4 user won't immediately hit a fault.